File: Presentacion/ClientApp/src/app/administrador/class/vuelo.ts
import { Ciudad, Aerolinea, Estado } from '../interfaces/interfaces';
export class Vuelo {
    numeroVuelo?: number;
    ciudadOrigenId?: number;
    ciudadDestinoId?: number;
    fecha?: string;
    horaSalida?: string;
    horaLlegada?: string;
    aerolineaId?: number;
    estadoVueloId?: number;
    ciudadOrigen?: Ciudad;
    ciudadDestino?: Ciudad;
    aerolinea?: Aerolinea;
    estadoVuelo?: Estado
}

File: Entidades/Vuelo.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Entidades
{
    public class Vuelo
    {
        [Key]
        public int NumeroVuelo { get; set; }
        
        [ForeignKey("Ciudad")]
        public int? CiudadOrigenId { get; set; }
        public Ciudad? CiudadOrigen { get; set; }

        [ForeignKey("Ciudad")]
        public int? CiudadDestinoId { get; set; }
        public Ciudad? CiudadDestino { get; set; }
        public DateTime Fecha { get; set; }
        public DateTime HoraSalida { get; set; }
        public DateTime HoraLlegada { get; set; }

        [ForeignKey("Aerolinea")]
        public int? AerolineaId { get; set; }
        public Aerolinea? Aerolinea { get; set; }

        [ForeignKey("Estado")]
        public int? EstadoVueloId { get; set; }
        public Estado? EstadoVuelo { get; set; }
    }
}